### PR TITLE
*Ruby Group Capstone Project - catalog*

### PR DESCRIPTION
**Pull request by ClarenceAtim**
*Ruby Group Capstone Project - catalog*

Pull Request Description:
This pull request addresses the division of tasks and the implementation of specific features for our Ruby group capstone project, "Catalog of my things." We have separated the requirements into individual tasks and assigned them to team members. Here's a breakdown of the changes:

*Group Task:*
- Created an `Item` class in a separate .rb file.
- Defined and set up all `Item` class properties as mentioned in the diagram.
- Added methods as specified:
  - `can_be_archived?()` in the `Item` class, which returns true if the `published_date` is older than 10 years.
  - `move_to_archive()` in the `Item` class, which reuses the `can_be_archived?()` method and changes the `archived` property to true if the result is true.

*Team Member #1 (Books and Labels):*
- Created a `Book` class in a separate .rb file.
- Created a `Label` class with an association to the `Item` class in a separate .rb file.
- Defined and set up all `Book` class and `Label` class properties as mentioned in the diagram.
- Implemented methods as specified:
  - `add_item` method in the `Label` class to add an instance of the `Item` class to the collection of items.
  - `can_be_archived?()` in the `Book` class, which overrides the method from the parent class and returns true if the parent's method returns true OR if `cover_state` equals "bad."
- Added unit tests for all implemented methods.

*Team Member #2 (MusicAlbums and Genres):*
- Created a `MusicAlbum` class in a separate .rb file.
- Created a `Genre` class with an association to the `Item` class in a separate .rb file.
- Defined and set up all `MusicAlbum` class and `Genre` class properties as mentioned in the diagram.
- Implemented methods as specified:
  - `add_item` method in the `Genre` class to add an instance of the `Item` class to the collection of items.
  - `can_be_archived?()` in the `MusicAlbum` class, which overrides the method from the parent class and returns true if the parent's method returns true AND if `on_spotify` equals true.
- Added unit tests for all implemented methods.

*Team Member #3 (Games and Authors):*
- Created a `Game` class in a separate .rb file.
- Created an `Author` class with an association to the `Item` class in a separate .rb file.
- Defined and set up all `Game` class and `Author` class properties as mentioned in the diagram.
- Implemented methods as specified:
  - `add_item` method in the `Author` class to add an instance of the `Item` class to the collection of items.
  - `can_be_archived?()` in the `Game` class, which overrides the method from the parent class and returns true if the parent's method returns true AND if `last_played_at` is older than 2 years.
- Added unit tests for all implemented methods.

Additionally, we have ensured data preservation by saving collections in .json files and created a schema.sql file with tables that mirror the structure of the classes we've created.

### CODE REVIEW

File: Modules/main_module.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  def add_book(app)
+    puts "\n"
+    puts 'Who is the publisher [Enter Publisher Name]:'
+    publisher = gets.chomp
+    puts 'What Is the cover\'s state like? [Enter good or bad]:'
+    cover_state = gets.chomp
+    puts 'When was the book published [Enter date format YYYY-MM-DD]:'
+    publish_date = gets.chomp
+    app.create_book(publisher, cover_state, publish_date)
+  end
```
Comment (fobadara) [inline Modules/main_module.rb]: [REQUIRED] I like the fact that you found a workaround for the scenario where a user enters an invalid date in most places in your app but you missed the point where the user tries to add a book. Have a look at the screenshot below:
image

Please prevent this error from happening

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi team, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

REDO

## Highlights
✅ Descriptive README
✅ Descriptive PR summary
✅ Clean code
✅ Good first attempt
✅ Good implementation of the principle of separating concerns.
✅ Good project structure
✅ Good understanding of Ruby and the project requirements
✅ Nice design

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi Team,
# STATUS: APPROVED! 🥇 💯 
done

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Review by ridwanediallo — COMMENTED** [review]
# STATUS: APPROVED 🟢
## Hi team, 
Congratulations!GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow correctly. ✔
- Your readme file looks professional and descriptive. ✔


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**